optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -240,12 +393,14 @@ def forward(
             else:
                 outputs = block(
                     hidden_states,
-                    layer_past=layer_past,
+                    position_bias=alibi,
                     attention_mask=causal_mask,
+                    layer_past=layer_past,
                     use_cache=use_cache,
                     output_attentions=output_attentions,
-                    position_bias=alibi,
                     token_idx=token_idx,
+                    reuse_cache=reuse_cache,
+                    cache_idx=cache_idx,
                 )
 
             hidden_states = outputs[0]
```

Comment:
@atakaha can you also fix the line `if use_cache is True:` to `if use_cache:`

Reply:
do you mean like this?
```python
                    reuse_cache=reuse_cache if use_cache else False,
                    cache_idx=cache_idx if use_cache else None,
                 )
```